RecoParticleFlow/PFClusterProducer/plugins/Basic2DGenericPFlowPositionCalc.h

```diff
@@ -8,16 +8,58 @@
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 
 #include "RecoParticleFlow/PFClusterProducer/interface/CaloRecHitResolutionProvider.h"
+#include <tuple>
 
 class Basic2DGenericPFlowPositionCalc : public PFCPositionCalculatorBase {
  public:
   Basic2DGenericPFlowPositionCalc(const edm::ParameterSet& conf) :
     PFCPositionCalculatorBase(conf),    
     _posCalcNCrystals(conf.getParameter<int>("posCalcNCrystals")),
-    _logWeightDenom(1./conf.getParameter<double>("logWeightDenominator")),
     _minAllowedNorm(conf.getParameter<double>("minAllowedNormalization"))
-
   {  
+
+    std::vector<int> detectorEnum;
+    std::vector<int> depths;
+    std::vector<double> logWeightDenom;
+    std::vector<float> logWeightDenomInv;
+
+    if(conf.exists("logWeightDenominatorByDetector")) {
+
+      const std::vector<edm::ParameterSet>& logWeightDenominatorByDetectorPSet = conf.getParameterSetVector("logWeightDenominatorByDetector");
+
+      for( const auto& pset : logWeightDenominatorByDetectorPSet ) {
+
+	if(!pset.exists("detector")) continue;
```

Comment:
would this even be a legal PSet without a "detector"?
This looks like a case for an exception.

Reply:
I will add

Reply:
just dropping this line will make an exception in the following line